mne/channels/layout.py

```diff
@@ -971,3 +972,76 @@ def apply_montage(info, montage):
         raise ValueError('None of the sensors defined in the montage were '
                          'found in the info structure. Check the channel '
                          'names.')
+
+
+def generate_2d_layout(xy, w=.07, h=.05, pad=.02, ch_names=None,
+                       ch_indices=None, name='ecog', bg_image=None):
+    '''Generates a 2-D layout for plotting with plot_topo methods and
+    functions. XY points will be normalized between 0 and 1, where
+    normalization extremes will be either the min/max of xy, or
+    the width/height of bg_image.
+
+    Parameters
+    -------
+    xy : ndarray (N x 2)
+        The xy coordinates of sensor locations.
+    w : float
+        The width of each sensor's axis (between 0 and 1)
+    h : float
+        The height of each sensor's axis (between 0 and 1)
+    pad : float
+        Portion of the box to reserve for padding. The value can range between
+        0.0 (boxes will touch, default) to 1.0 (boxes consist of only padding).
+    ch_names : list
+        The names of each channel. Must be a list of strings, with one
+        string per channel.
+    ch_indices : list
+        Index of each channel - must be a collection of unique integers,
+        one index per channel.
+    name : string
+        The name of this layout type.
+    bg_image : str | ndarray
+        The image over which sensor axes will be plotted. Either a path to an
+        image file, or an array that can be plotted with plt.imshow. If
+        provided, xy points will be normalized by the width/height of this
+        image. If not, xy points will be normalized by their own min/max.
+
+    Returns
+    --------
+    layout : Layout
+        A Layout object that can be plotted with plot_topo
+        functions and methods.
+    '''
+
+    if ch_indices is None:
+        ch_indices = np.arange(xy.shape[0])
+    if ch_names is None:
+        ch_names = ['{0}'.format(i) for i in ch_indices]
+
+    assert len(ch_names) == len(ch_indices), 'Unequal # ch names and indices'
+    assert len(ch_names) == len(xy), 'Unequal # ch names and xy vals'
```

Comment:
Ahh I see - thanks for the clarification

On Thu, Apr 16, 2015 at 11:25 AM, Eric Larson <notifications@github.com>
wrote:

> In mne/channels/layout.py
> <https://github.com/mne-tools/mne-python/pull/1762#discussion_r28536215>:
>
> > +        image. If not, xy points will be normalized by their own min/max.
> > +
> > +    Returns
> > +    --------
> > +    layout : Layout
> > +        A Layout object that can be plotted with plot_topo
> > +        functions and methods.
> > +    '''
> > +
> > +    if ch_indices is None:
> > +        ch_indices = np.arange(xy.shape[0])
> > +    if ch_names is None:
> > +        ch_names = ['{0}'.format(i) for i in ch_indices]
> > +
> > +    assert len(ch_names) == len(ch_indices), 'Unequal # ch names and indices'
> > +    assert len(ch_names) == len(xy), 'Unequal # ch names and xy vals'
>
> no those are only used in testing, in the main code we always follow this
> form:
>
> if bad_condition:
>     raise AppropriateError('you did something bad')
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/1762/files#r28536215>.
>